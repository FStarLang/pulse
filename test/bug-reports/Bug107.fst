module Bug107
#lang-pulse

open Pulse.Lib.Pervasives

val foo : int -> int -> slprop
let foo x y = emp


fn test0 ()
  requires foo 1 2
  ensures emp
{
  unfold foo 1 2;
  ()
}



fn test1 ()
  requires foo 1 2
  ensures emp
{
  unfold foo;
  ()
}



fn test2 ()
  requires foo 1 2
  ensures emp
{
  unfold foo 1;
  ()
}


fn test3 ()
  requires foo 1 2
  ensures emp
{
  unfold foo 1 _;
  ()
}


fn test4 ()
  requires foo 1 2
  ensures emp
{
  unfold foo _ 2;
  ()
}

