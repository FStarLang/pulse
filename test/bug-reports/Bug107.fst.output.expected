* Info at Bug107.fst(45,9-45,16):
  - Expected failure:
  - Could not resolve implicit uu___#86 of type Prims.int in term Bug107.foo 1 _

* Info at Bug107.fst(56,9-56,16):
  - Expected failure:
  - Could not resolve implicit uu___#86 of type Prims.int in term Bug107.foo _ 2

